azkaban-execserver/src/main/java/azkaban/execapp/event/JobCallbackManager.java

```diff
@@ -0,0 +1,267 @@
+    return callbackMbean;
+  }
+
+  @Override
+  public void handleEvent(Event event) {
+    if (!isInitialized) {
+      return;
+    }
+
+    if (event.getRunner() instanceof JobRunner) {
+      try {
+        if (event.getType() == Event.Type.JOB_STARTED) {
+          processJobCallOnStart(event);
+        } else if (event.getType() == Event.Type.JOB_FINISHED) {
+          processJobCallOnFinish(event);
+        }
+      } catch (Throwable e) {
+        // Use job runner logger so user can see the issue in their job log
+        JobRunner jobRunner = (JobRunner) event.getRunner();
+        jobRunner.getLogger().error(
+            "Encountered error while hanlding job callback event", e);
+      }
+    } else {
+      logger.warn("((( Got an unsupported runner: "
+          + event.getRunner().getClass().getName() + " )))");
+    }
+
+  }
+
+  private void processJobCallOnFinish(Event event) {
+    JobRunner jobRunner = (JobRunner) event.getRunner();
+
+    if (!JobCallbackUtil.isThereJobCallbackProperty(jobRunner.getProps(),
+        ON_COMPLETION_JOB_CALLBACK_STATUS)) {
+      return;
+    }
+
+    // don't want to waste time resolving properties if there are no
+    // callback properties to parse
+    Props props = PropsUtils.resolveProps(jobRunner.getProps());
+
+    Map<String, String> contextInfo =
+        JobCallbackUtil.buildJobContextInfoMap(event, azkabanHostName);
+
+    JobCallbackStatusEnum jobCallBackStatusEnum = null;
+    Logger jobLogger = jobRunner.getLogger();
+
+    Status jobStatus = jobRunner.getNode().getStatus();
+
+    if (jobStatus == Status.SUCCEEDED) {
+
+      jobCallBackStatusEnum = JobCallbackStatusEnum.SUCCESS;
+
+    } else if (jobStatus == Status.FAILED
+        || jobStatus == Status.FAILED_FINISHING || jobStatus == Status.KILLED) {
+
+      jobCallBackStatusEnum = JobCallbackStatusEnum.FAILURE;
+    } else {
+      jobLogger.info("!!!! WE ARE NOT SUPPORTING JOB CALLBACKS FOR STATUS: "
+          + jobStatus);
+      jobCallBackStatusEnum = null; // to be explicit
+    }
+
+    String jobId = contextInfo.get(CONTEXT_JOB_TOKEN);
+
+    if (jobCallBackStatusEnum != null) {
+      List<HttpRequestBase> jobCallbackHttpRequests =
+          JobCallbackUtil.parseJobCallbackProperties(props,
+              jobCallBackStatusEnum, contextInfo, maxNumCallBack, jobLogger);
+
+      if (!jobCallbackHttpRequests.isEmpty()) {
+        String msg =
+            String.format("Making %d job callbacks for status: %s",
+                jobCallbackHttpRequests.size(), jobCallBackStatusEnum.name());
+        jobLogger.info(msg);
+
+        addDefaultHeaders(jobCallbackHttpRequests);
+
+        JobCallbackRequestMaker.getInstance().makeHttpRequest(jobId, jobLogger,
+            jobCallbackHttpRequests);
+      } else {
+        jobLogger.info("No job callbacks for status: " + jobCallBackStatusEnum);
+      }
+    }
+
+    // for completed status
+    List<HttpRequestBase> httpRequestsForCompletedStatus =
```

Comment:
This set of code is the duplicate of the code above? Maybe they can be combined?